src/lib.rs

```diff
@@ -231,30 +231,25 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 /// Note that when following symbolic/soft links, loops are detected and an
 /// error is reported.
 #[derive(Debug)]
-pub struct WalkDir {
-    opts: WalkDirOptions,
+pub struct WalkDir<'slf> {
```

Comment:
This is a breaking change. And this should also reveal why the `WalkDir` type requires `'static`: so that it isn't itself generic.

This also strikes me as a very odd use of lifetimes and I don't know what kinds of implications it will have on downstream APIs.

Reply:
Heh, well _most_ use of lifetimes feels strange to me still... I was hoping in the process of implementing the change and getting your input I'd come to understand whether/how the implications were problematic, if they didn't happen to be trivially close to something workable.